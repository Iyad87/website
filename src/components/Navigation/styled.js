import GatsbyLink from 'gatsby-link';
import styled from 'styled-components';

import { text } from '../../styles/colours';

export const List = styled.ul`
  list-style-type: none;
  margin: 0;
  padding-left: 0;
`;

export const Item = styled.li`
  display: inline-block;
`;

export const Link = styled(GatsbyLink)`
  color: ${text};
  display: inline-block;
  font-size: 1.1em;
  padding: 10px 15px;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;

export const SecondLastLink = Link.extend`
  padding-right: 30px;
`;

export const LastLink = Link.extend`
  border: 1px solid #ddd;
  border-radius: 4px;
  padding: 10px 30px;
`;
