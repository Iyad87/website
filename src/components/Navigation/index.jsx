import React from 'react';
import { Link, LastLink, SecondLastLink, List, Item } from './styled';

export default function Navigation() {
  return (
    <List>
      <Item>
        <Link to="/case-studies">Case Studies</Link>
      </Item>
      <Item>
        <Link to="/services">Services</Link>
      </Item>
      <Item>
        <Link to="/community">Community</Link>
      </Item>
      <Item>
        <SecondLastLink to="/about-us">About Us</SecondLastLink>
      </Item>
      <Item>
        <LastLink to="/join-us">Join Us</LastLink>
      </Item>
    </List>
  );
}
