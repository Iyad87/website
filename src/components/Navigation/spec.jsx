import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './index';

test('components/Navigation snapshot', () => {
  const wrapper = shallow(<Navigation />);

  expect(wrapper).toMatchSnapshot();
});
