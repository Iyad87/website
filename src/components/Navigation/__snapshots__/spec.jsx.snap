// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation snapshot 1`] = `
<nav>
  <styled__List>
    <styled__Item>
      <styled__Link
        href="/case-studies"
      >
        Case Studies
      </styled__Link>
    </styled__Item>
    <styled__Item>
      <styled__Link
        href="/services"
      >
        Services
      </styled__Link>
    </styled__Item>
    <styled__Item>
      <styled__Link
        href="/community"
      >
        Community
      </styled__Link>
    </styled__Item>
    <styled__Item>
      <styled__Link
        href="/about-us"
        padRight={true}
      >
        About Us
      </styled__Link>
    </styled__Item>
    <styled__Item>
      <styled__Link
        href="/join-us"
      >
        Join Us
      </styled__Link>
    </styled__Item>
  </styled__List>
</nav>
`;
